### PR TITLE
[add] pollard rho and pollard p-1 factorization methods

### LINKED ISSUE #1796: Add Pollard Rho and Pollard P-1 factorization methods

### PR DESCRIPTION
**Pull request by sandy9999**
Fixes #1796 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [ ] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added the Pollard Rho and Pollard p-1 factorization methods

#### Languages Used:
- C++

#### Files Added:
- Pollard_Rho_Factorization/pollard_rho.cpp
- Pollard_Pminusone_Factorization/pollard_pminus1.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 .DS_Store
-.vscode/*
\ No newline at end of file
+.vscode/*
+a.out
+./*/a.out
```
Comment (SakshiShreya) [inline .gitignore]: Please remove this file from the commit.

File: Pollard_Pminusone_Factorization/pollard_pminus1.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    return -1;
+}
+
+
+int main()
+{
+    ll n;
+    cin>>n;
+    cout<<"One of the factors of "<<n<<" is "<<pollard_p1(n,8);
+           return 0;
```
Comment (SakshiShreya) [inline Pollard_Pminusone_Factorization/pollard_pminus1.cpp]: Use 4 spaces only in this line.

File: Pollard_Rho_Factorization/pollard_rho.cpp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+    ll n;
+    cin>>n;
+    cout<<"One of the factors of "<<n<<" is "<<pollard_rho(n);
+           return 0;	
+}
+
+/*Sample input
+n = 10967535067
+*/
```
Comment (SakshiShreya) [inline Pollard_Rho_Factorization/pollard_rho.cpp]: Add a new line at the end of file.

File: Pollard_Pminusone_Factorization/pollard_pminus1.cpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    ll n;
+    cin>>n;
+    cout<<"One of the factors of "<<n<<" is "<<pollard_p1(n,8);
+           return 0;	
+}
+
+/*
+Sample Input
+n = 57247159
+
```
Comment (SakshiShreya) [inline Pollard_Pminusone_Factorization/pollard_pminus1.cpp]: Mention the output also.

File: Pollard_Pminusone_Factorization/pollard_pminus1.cpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
+#include <bits/stdc++.h>
```
Comment (SakshiShreya) [inline Pollard_Pminusone_Factorization/pollard_pminus1.cpp]: Add documentation at the top.

File: Pollard_Rho_Factorization/pollard_rho.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#include <bits/stdc++.h>
```
Comment (SakshiShreya) [inline Pollard_Rho_Factorization/pollard_rho.cpp]: Add documentation at the top.

File: Pollard_Rho_Factorization/pollard_rho.cpp [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+int main()
+{
+    ll n;
+    cin>>n;
+    cout<<"One of the factors of "<<n<<" is "<<pollard_rho(n);
+           return 0;	
+}
+
+/*Sample input
+n = 10967535067
```
Comment (SakshiShreya) [inline Pollard_Rho_Factorization/pollard_rho.cpp]: Mention the output also.

**Comment by jainaman224** [comment]
@sandy9999 You have deleted the .gitignore file. Instead you should revert the changes to master in that file. Please seek help from `git_github` channel on slack.

**Comment by somya-kapoor** [comment]
@sandy9999 Are you working on it?

**Comment by somya-kapoor** [comment]
@sandy9999 Are you working on it?

**Comment by somya-kapoor** [comment]
@sandy9999 Due to no activities for more than 7 days, I am closing this PR.

**Pull request closed without merging**